text/0000-arguments.md

```diff
@@ -0,0 +1,195 @@
+args = ["ARG1", "ARG2"]
+```
+we currently execute `bash -c "<CMD>" "ARG1" "ARG2"`. This makes it hard to use arguments intuitively in a shell process.
+For example,
+```
+[[process]]
+command = "echo"
+args = ["hello", "world"]
+```
+does not print `hello world`. Instead users must construct a string command that includes arguments
+```
+[[process]]
+command = "echo hello world"
+args = []
+```
+This already causes confusion for users (https://github.com/buildpacks/lifecycle/issues/281). The problem is compounded if we allow users to supply additional arguments. We cannot append arguments to the command string in a reliable way and constructing a command that accepts args in the way they are provided is confusing and difficult (buildpack authors likely will not do this correctly).
+
+The RFC proposes and changes to the launchers handling of args to make the execution of the following process
+```
+[[process]]
+command = "<CMD>"
+args = ["ARG1", "ARG2"]
+```
+produce results equivalent to the execution of `"<CMD>" "<ARG1>" "<ARG2>"` in a shell
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Terminology
+* shell process - a process with `direct=false`
+
+## Philosophy
+The goal is to make the runtime usage of a CNB container as similar as possible to the runtime usage on a non-CNB container, without removing functionality or violating assumptions that have been firmly established in the buildpack community (e.g. defaulting to `web`).
+
+Because the `launcher` must occupy the container entrypoint we will instead use `CNB_PROCESS_TYPE` as a CNB-specific analog to the entrypoint. Users can already set `CNB_PROCESS_TYPE` to toggle between processes. Setting `CNB_PROCESS_TYPE=override` will be analogous to clearing the entrypoint.
+
+## Behavior
+### Selecting a process type at runtime
+Users will no longer be able to pass a process-type to the launcher, instead they must set `CNB_PROCESS_TYPE` in the running container to select a process type.
+If `CNB_PROCESS_TYPE` is unset, it will continue to default to `web`.
+
+This functionality has existed for a long time, but we will remove the second way to set toggle the same behavior, by passing a single argument containing a process-type to the launcher (i.e. `docker run <image> my-process-type` won't work anymore).
+
+### Specifying a custom process at runtime
+If users wish to provide a custom command instead of using one of the buildpack generated process types, they must set `CNB_PROCESS_TYPE=override` to explicitly indicate that they wish to override the entire command, rather than to append args.
+
+Running `docker run --env CNB_PROCESS_TYPE=override <image> echo hello world`, will be equivalent to running a process type where `direct = false`, `command = "echo"` and `args = ["hello", "world"]`
+
+Running `docker run --env CNB_PROCESS_TYPE=override <image> -- echo hello world`, will be equivalent to running a process type where `direct = true`, `command = "echo"` and `args = ["hello", "world"]`
```

Comment:
This `--` indicates `direct = true` sort of pops up out of nowhere.  What's the background/prior art/justification for this syntax?

Reply:
It is the current syntax for specifying a `direct=true` command at runtime

Reply:
Is that documented anywhere, or just implemented?

Reply:
It's in the Buildpack API spec -https://github.com/buildpacks/spec/blob/master/buildpack.md#launch

imho the override behavior here belongs in the Platform API